src/users/users.service.ts

```diff
@@ -115,7 +116,24 @@ export class UsersService {
    */
   async deleteUser(id: number): Promise<void> {
     Logger.log(`[UsersService] deleteUser(${id})`);
+    const detailedUser: User = await this.getUserById(id);
+
+    if (!detailedUser) {
+      throw new NotFoundException();
+    }
+
+    if (detailedUser.parcels) {
+      if (detailedUser.parcels.find(parcel =>
+        parcel.parcelTrackingStatus === ParcelStatus.distribution ||
+        parcel.parcelTrackingStatus === ParcelStatus.exception)) {
+        throw new BadRequestException(
+          `User with id ${id} has parcels in distribution or exception status assigned, and therefore can not be deleted`,
+        );
+      }
+    }
+
     const user: User = await this.userRepository.findOne({ id });
```

Comment:
I had to fetch the user again, since the "detailedUser" has a parcels field, and to update it.
I'm not familiar with TypeORM too much to know if it's possible to do it in a better way, and avoid this http request, do you know any?